source/exception.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/config.hpp>
+#include <unifex/exception.hpp>
+
+namespace unifex {
+namespace _except_ptr {
+#if !UNIFEX_HAS_FAST_MAKE_EXCEPTION_PTR()
+[[noreturn]] inline void _ref::rethrow() const {
+  throw_(p_);
+  UNIFEX_ASSUME_UNREACHABLE;
+}
+
+[[nodiscard]] std::exception_ptr _fn::operator()(_ref const e) const noexcept {
```

Comment:
Is the `[[nodiscard]]` attribute needed on the definition or only on the declaration?

Reply:
Don't know. I'll try and see.